src/insert_completer.cc

```diff
@@ -348,7 +348,7 @@ InsertCompletion complete_line(const SelectionList& sels,
     const ColumnCount tabstop = options["tabstop"].get<int>();
     const ColumnCount column = get_column(buffer, tabstop, cursor_pos);
 
-    StringView prefix = trim_indent(buffer[cursor_pos.line].substr(0_byte, cursor_pos.column));
+    String prefix = trim_indent(buffer[cursor_pos.line].substr(0_byte, cursor_pos.column));
```

Comment:
Those bugs are quite bad, I really wish there would be a good way in C++ to avoid that with some kind of life time annotations.

Reply:
Same as you, I really wish C++ could trigger a warning about type mismatch in this case...

Reply:
Well, the different types are fine, Strings can convert to StringView, but I wish we could have something like this [proposal](http://wg21.link/p0936r0) to specify that using that StringView past the lifetime of the original String is incorrect.